src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_attend_and_excite.py

```diff
@@ -0,0 +1,1033 @@
+            batch_size * num_images_per_prompt,
+            num_channels_latents,
+            height,
+            width,
+            prompt_embeds.dtype,
+            device,
+            generator,
+            latents,
+        )
+
+        # 6. Prepare extra step kwargs. TODO: Logic should ideally just be moved out of the pipeline
+        extra_step_kwargs = self.prepare_extra_step_kwargs(generator, eta)
+
+        # default config for step size from original repo
+        scale_range = np.linspace(1.0, 0.5, len(self.scheduler.timesteps))
+
+        if max_iter_to_alter is None:
+            max_iter_to_alter = len(self.scheduler.timesteps) + 1
+
+        # 7. Denoising loop
+        num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+        with self.progress_bar(total=num_inference_steps) as progress_bar:
+            for i, t in enumerate(timesteps):
+                with torch.enable_grad():
+                    latents = latents.clone().detach().requires_grad_(True)
+
+                    # Forward pass of denoising with text conditioning
+                    noise_pred_text = self.unet(
+                        latents,
+                        t,
+                        encoder_hidden_states=prompt_embeds[1].unsqueeze(0),
+                        cross_attention_kwargs=cross_attention_kwargs,
+                    ).sample
+                    self.unet.zero_grad()
+
+                    # Get max activation value for each subject token
+                    max_attention_per_index = self._aggregate_and_get_max_attention_per_token(
+                        indices_to_alter=indices_to_alter,
+                    )
+
+                    loss = self._compute_loss(max_attention_per_index=max_attention_per_index)
+
+                    del noise_pred_text
+                    torch.cuda.empty_cache()
+
+                    # If this is an iterative refinement step, verify we have reached the desired threshold for all
+                    if i in thresholds.keys() and loss > 1.0 - thresholds[i]:
+                        loss, latents, max_attention_per_index = self._perform_iterative_refinement_step(
+                            latents=latents,
+                            indices_to_alter=indices_to_alter,
+                            loss=loss,
+                            threshold=thresholds[i],
+                            text_embeddings=prompt_embeds,
+                            text_input=text_inputs,
+                            step_size=scale_factor * np.sqrt(scale_range[i]),
+                            t=t,
+                        )
+
+                    # Perform gradient update
+                    if i < max_iter_to_alter:
+                        if loss != 0:
+                            latents = self._update_latent(
+                                latents=latents, loss=loss, step_size=scale_factor * np.sqrt(scale_range[i])
+                            )
```

Comment:
```suggestion
        scale_range = np.linspace(1.0, 0.5, len(self.scheduler.timesteps))
        step_size = scale_factor * np.sqrt(scale_range)

        if max_iter_to_alter is None:
            max_iter_to_alter = len(self.scheduler.timesteps) + 1

        # 7. Denoising loop
        num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
        with self.progress_bar(total=num_inference_steps) as progress_bar:
            for i, t in enumerate(timesteps):
                with torch.enable_grad():
                    latents = latents.clone().detach().requires_grad_(True)

                    # Forward pass of denoising with text conditioning
                    noise_pred_text = self.unet(
                        latents,
                        t,
                        encoder_hidden_states=prompt_embeds[1].unsqueeze(0),
                        cross_attention_kwargs=cross_attention_kwargs,
                    ).sample
                    self.unet.zero_grad()

                    # Get max activation value for each subject token
                    max_attention_per_index = self._aggregate_and_get_max_attention_per_token(
                        indices_to_alter=indices_to_alter,
                    )

                    loss = self._compute_loss(max_attention_per_index=max_attention_per_index)

                    del noise_pred_text
                    torch.cuda.empty_cache()

                    # If this is an iterative refinement step, verify we have reached the desired threshold for all
                    if i in thresholds.keys() and loss > 1.0 - thresholds[i]:
                        loss, latents, max_attention_per_index = self._perform_iterative_refinement_step(
                            latents=latents,
                            indices_to_alter=indices_to_alter,
                            loss=loss,
                            threshold=thresholds[i],
                            text_embeddings=prompt_embeds,
                            text_input=text_inputs,
                            step_size=step_size[i],
                            t=t,
                        )

                    # Perform gradient update
                    if i < max_iter_to_alter:
                        if loss != 0:
                            latents = self._update_latent(
                                latents=latents, loss=loss, step_size=step_size[i]
                            )
```
nit

Reply:
@williamberman what happened here? 😂

Reply:
diff render messed up haha

was recommending putting these next to each other
```py
        scale_range = np.linspace(1.0, 0.5, len(self.scheduler.timesteps))
        step_size = scale_factor * np.sqrt(scale_range)
```

and using 

```py
step_size=step_size[i]
```

in locations where it was using

```py
step_size=scale_factor * np.sqrt(scale_range[i])
```